### PR TITLE
Initial implementation of Prometheus server

### PR DESCRIPTION
**Pull request by tgross**
@misterbisson this is an in-progress PR for our Prometheus server.

I'm using `consul-template` to re-render the Prometheus configuration file in order to update targets for any container that's advertising on a service named `metrics`. Includes setup script, README, and both local/remote compose file. This doesn't yet include a demonstration metrics target.

The current build has been pushed to the Hub as [autopilotpattern/prometheus:latest](https://hub.docker.com/r/autopilotpattern/prometheus/).

cc @misterbisson

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,43 @@
...
+prometheus_prometheus_1      /bin/containerbuddy   Up   0.0.0.0:9090->9090/tcp
+                             /bin/prometheus...
+```
+
+
+Once you have Prometheus running you should be able to check its current status by making an HTTP request to its own metrics endpoint:
+
+
+```bash
+$ prometheus=$(triton inst get prometheus_prometheus_1 | json -a ips.1)
```
Comment (misterbisson) [inline README.md]: `triton ip prometheus_prometheus_1` also works, though I don't know what the `ip` method was added.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,43 @@
...
+$ curl "http://${prometheus}:9090/metrics" | less # there's a lot of data!
```
Comment (misterbisson) [inline README.md]: `curl "http://$(triton ip prometheus_prometheus_1):9090/metrics" | less`, maybe?

**Comment by misterbisson** [comment]
Tested; works for me.

:house_with_garden: :walking:

File: README.md (hunk 2 above)
Comment (tgross) [inline README.md]: Cool. Will do.

File: etc/containerbuddy.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "consul": "{{ .CONSUL }}:8500",
+  "services": [
+    {
+      "name": "prometheus",
+      "port": 9090,
+      "health": ["true"],
```
Comment (tgross) [inline etc/containerbuddy.json]: TODO: replace this by `curl -so /dev/null localhost:9090`

**Comment by tgross** [comment]
Made those changes. Should we merge this in and do a demo app under separate PR?

**Comment by misterbisson** [comment]
Yep, it makes sense to merge this one now.

**Pull request merged**